lib/libutee/arch/arm/utee_syscalls_a64.S

```diff
@@ -46,3 +46,4 @@
 
 #include "utee_syscalls_asm.S"
 
+emit_aarch64_feature_1_and	GNU_PROPERTY_AARCH64_FEATURE_1_BTI
```

Comment:
I don't like the fact that this is introduced in a random assembly file :-/ Could we have one file per library e.g., `bti_a64.S` which would be compiled in only when BTI is enabled?

Reply:
I have added these in all a64 files. I will check about obe file per library part, if that works.

Reply:
I re-checked. The section needs to be present in each of the object files for the compiler to finally add this section in the generated elf. So, the macro will need to be added to the assembly files.

One way to figure out which assembly files need the section to be added is to use "-z force-bti" as link-ldflags. Doing so, compiler would forcefully add .not.gnu.property section in final elf and give out warnings on which .o files had this property missing.

Reply:
Ah yes, that makes sense since one doesn't want BTI enabled unless all code has proper support for it.
For clarity, may I suggest using the `BTI()` macro I proposed?
```
BTI(emit_aarch64_feature_1_and GNU_PROPERTY_AARCH64_FEATURE_1_BTI)
```
It would be more obvious that the thing is emitted only when BTI is enabled.

One other thing: when BTI is enabled, perhaps we should have a post-link step that checks that the feature is indeed present in the ELF binary. I'm afraid we could easily miss adding `the emit_aarch64_feature_1_and` line in new assembly files.